python-spec/src/somacore/data.py

```diff
@@ -195,8 +195,18 @@ def create(
         :param uri: The URI where the array will be created.
         :param type: The Arrow type to store in the array.
             If the type is unsupported, an error will be raised.
-        :param shape: The length of each dimension as a sequence,
-            e.g. ``(100, 10)``. All lengths must be in the postive int64 range.
+        :param shape: The maximum capacity of each dimension, including room
+            for any intended future appends, as a sequence.  E.g. ``(100, 10)``.
+            All lengths must be in the postive int64 range, or ``None``.
+
+            For `SOMASparseNDArray` only, if a slot is None, then the maximum
+            possible int64 will be used.  This makes a `SOMASparseNDArray`
+            growable.
+
+            It's necessary to say `shape=(None, None)` or `shape=(None, None, None)`
+            rather than more simply `shape=None` since the first two are how one
+            specifies N=2 or N=3, respectively, for N-dimensional arrays: the length
+            of this sequence is the number of dimensions of the array.
```

Comment:
I would put this part in the first paragraph, since it clarifies the length of input sequence–dimensionality of array relationship